include/students.php

```diff
@@ -0,0 +1,153 @@
+<?php
+
+// --------------------------------------------------------------------------
+//This function accepts allStudent array as reference and adds to the array 
+//the remaining "credits" the student have 
+// --------------------------------------------------------------------------
+
+function addRemainingCredits(&$allStudents){                                          
+      
+    $creditsAmount = getCreditsAmount();                                              //Gets the amount of credit the students have from the credits table
+    $classesAmount = getClassesAmount();                                              //Same with classes amount
+
+    $studentIdsArray = array_column($classesAmount, 'StudentId');                   //Gets an array of studentsId from the classes array because is the easiest way I found to search for specific studentId.
+  
+    $i = 0;
+    foreach($creditsAmount as $credit){                                              //Go throught the creditsAmount array and
+        if (in_array($creditsAmount[$i]['StudentId'], $studentIdsArray)){           //branch if the studentsId from the credits array is found in the classes array. I do this because 
```

Comment:
Ah interesting! Let me see! The studentIdsArray here is populated from the classAmount array (instead of credits amount).  I think I will make the name less generic. So, as I loop through the credits array, I check if the current student is in the class array(but using the ids array instead), and if is in there I loop through the class array to perform the calculation and update the credits.
The reason I'm using an array of ids instead of the class array directly is because the syntax was a bit confusing to me so I  got away with having an ids array to check.